RequestService/RequestService.Repo/Repository.cs

```diff
@@ -1082,6 +1098,69 @@ public List<ShiftJob> GetUserShiftJobsByFilter(GetUserShiftJobsByFilterRequest r
                 jobs = jobs.Where(x => request.JobStatusRequest.JobStatuses.Contains((JobStatuses)x.JobStatusId)).ToList();
             };
 
+            if (request.DateFrom.HasValue)
+            {
+                jobs = jobs.Where(x => x.NewRequest.Shift.StartDate >= request.DateFrom.Value).ToList();
+            }
+
+            if (request.DateTo.HasValue)
+            {
+                jobs = jobs.Where(x => x.NewRequest.Shift.StartDate.AddMinutes(x.NewRequest.Shift.ShiftLength) <= request.DateTo.Value).ToList();
+            }
+
+            return jobs.Select(x => new ShiftJob()
+            {
+                JobID = x.Id,
+                RequestID = x.NewRequest.Id,
+                Activity = (HelpMyStreet.Utils.Enums.SupportActivities)x.SupportActivityId,
+                JobStatuses = (JobStatuses)x.JobStatusId,
+                StartDate = x.NewRequest.Shift.StartDate,
+                ShiftLength = x.NewRequest.Shift.ShiftLength,
+                VolunteerUserID = x.VolunteerUserId
+            }).ToList();
+        }
+
+        public List<ShiftJob> GetOpenShiftJobsByFilter(GetOpenShiftJobsByFilterRequest request)
+        {
+            byte requestTypeShift = (byte)RequestType.Shift;
+            byte jobstatusOpen = (byte)JobStatuses.Open;
+            var jobs = _context.Job
+                .Include(i => i.NewRequest)
+                .ThenInclude(i => i.Shift)
+                .Include(i => i.JobAvailableToGroup)
+                .Where(x => x.NewRequest.RequestType == requestTypeShift 
+                && x.JobStatusId == jobstatusOpen).ToList();
+
+            if (jobs == null || jobs.Count == 0)
+            {
+                return null;
+            }
+
+            if (request.ExcludeSiblingsOfJobsAllocatedToUserID.HasValue)
+            {
+                //TODO
+            }
+
+            if (request.SupportActivities.SupportActivities.Count > 0)
```

Comment:
Does this match the behaviour in GetJobsByFilter?

The alternative would be:
* null --> Don't filter on this field
* empty list --> Do filter on this field (and therefore return nothing)

I'm happy for it to be either as written, or the alternative, whichever matches the implementation in GetJobsByFilter.

Same for Groups and Locations below.